### DISCUSSION TITLE
Links to why typescript chose to opt in for these weird quirks

### DISCUSSION THREAD

**Discussion by andredewaard** [mattpocock/ts-reset#129] [General]
Maybe its a good idea to have links (if the exist) to pages where typescript explains, or has a discussion, about why they chose to have this behaviour.
For example why did typescript choose to return any instead of unknown on a fetch request. If we know why they choose to do so we can better consider using this plugin or not.

Great plugin though and I really think im going to use this in any project going forward.

**Comment by ooker777** [comment]
a year has passed. Have you had any news

**Reply by andredewaard** [reply]
Nope, although we have some good news on TS 5.5 which makes this plugin a lit less necessary: https://www.totaltypescript.com/type-predicate-inference
